controllers/fluxdeployer_controller.go

```diff
@@ -0,0 +1,310 @@
+// +kubebuilder:rbac:groups=kustomize.toolkit.fluxcd.io,resources=kustomizations,verbs=get;list;watch;create;update;patch;delete
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *FluxDeployerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	const (
+		sourceKey = ".metadata.source"
+	)
+
+	if err := mgr.GetFieldIndexer().IndexField(context.TODO(), &v1alpha1.FluxDeployer{}, sourceKey, func(rawObj client.Object) []string {
+		obj, ok := rawObj.(*v1alpha1.FluxDeployer)
+		if !ok {
+			return []string{}
+		}
+		return []string{obj.Spec.SourceRef.Name}
+	}); err != nil {
+		return fmt.Errorf("failed setting index fields: %w", err)
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&deliveryv1alpha1.FluxDeployer{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(
+			&source.Kind{Type: &v1alpha1.Snapshot{}},
+			handler.EnqueueRequestsFromMapFunc(r.findObjects(sourceKey)),
+			builder.WithPredicates(SnapshotDigestChangedPredicate{}),
+		).
+		Complete(r)
+}
+
+func (r *FluxDeployerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	obj := &v1alpha1.FluxDeployer{}
+	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get deployer object: %w", err)
+	}
+
+	patchHelper, err := patch.NewHelper(obj, r.Client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Always attempt to patch the object and status after each reconciliation.
+	defer func() {
+		// Patching has not been set up, or the controller errored earlier.
+		if patchHelper == nil {
+			return
+		}
+
+		// Set status observed generation option if the object is stalled or ready.
+		if conditions.IsStalled(obj) || conditions.IsReady(obj) {
+			obj.Status.ObservedGeneration = obj.Generation
+		}
+
+		if err := patchHelper.Patch(ctx, obj); err != nil {
+			return
+		}
+	}()
+
+	return r.reconcile(ctx, obj)
+}
+
+func (r *FluxDeployerReconciler) reconcile(ctx context.Context, obj *v1alpha1.FluxDeployer) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("reconciling flux-deployer", "name", obj.GetName())
+
+	// get snapshot
+	snapshot, err := r.getSnapshot(ctx, obj)
+	if err != nil {
+		logger.Info("could not find source ref", "name", obj.Spec.SourceRef.Name)
+		return ctrl.Result{RequeueAfter: r.RetryInterval}, nil
+	}
+
+	// requeue if snapshot is not ready
+	if conditions.IsFalse(snapshot, meta.ReadyCondition) {
+		logger.Info("snapshot not ready yet", "snapshot", snapshot.Name)
+		return ctrl.Result{RequeueAfter: r.RetryInterval}, nil
+	}
+
+	snapshotRepo, err := ocm.ConstructRepositoryName(snapshot.Spec.Identity)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	snapshotURL := fmt.Sprintf("oci://%s/%s", r.RegistryServiceName, snapshotRepo)
```

Comment:
We have a `RepositoryURL` in the snapshot, isn't that this...? 🤔

Reply:
It's actually a bit different and doesn't have the oci prefix

Reply:
Ah, right.